submissions/ses3332002/dom/script.js

```diff
@@ -0,0 +1,257 @@
+      "Baby Be Mine",
+      "The Girl Is Mine",
+      "Thriller",
+      "Beat It",
+      "Billie Jean",
+      "Human Nature",
+      "P.Y.T. (Pretty Young Thing)",
+      "The Lady in My Life"
+    ],
+    "bgColor": 18
+  },
+  {
+    "title": "Bad (1987)",
+    "cover": "bad",
+    "songs": [
+      "Bad",
+      "The Way You Make Me Feel",
+      "Speed Demon",
+      "Liberian Girl",
+      "Just Good Friends",
+      "Another Part of Me",
+      "Man in the Mirror",
+      "I Just Can't Stop Loving You",
+      "Dirty Diana",
+      "Smooth Criminal",
+      "Leave Me Alone"
+    ],
+    "bgColor": 0
+  },
+  {
+    "title": "Dangerous (1991)",
+    "cover": "dangerous",
+    "songs": [
+      "Jam",
+      "Why You Wanna Trip on Me",
+      "In the Closet",
+      "She Drives Me Wild",
+      "Remember the Time",
+      "Can't Let Her Get Away",
+      "Heal the World",
+      "Black or White",
+      "Who Is It",
+      "Give in to Me",
+      "Will You Be There",
+      "Keep the Faith",
+      "Gone Too Soon",
+      "Dangerous"
+    ],
+    "bgColor": 29
+  },
+  {
+    "title": "HIStory: Past, Present and Future, Book I (1995)",
+    "cover": "history",
+    "songs": [
+      "Scream",
+      "They Don’t Care About Us",
+      "Stranger in Moscow",
+      "This Time Around",
+      "Earth Song",
+      "D.S.",
+      "Money",
+      "Come Together",
+      "You Are Not Alone",
+      "Childhood",
+      "Tabloid Junkie",
+      "2 Bad",
+      "HIStory",
+      "Little Susie",
+      "Smile"
+    ],
+    "bgColor": 332
+  },
+  {
+    "title": "Invincible (2001)",
+    "cover": "invincible",
+    "songs": [
+      "Unbreakable",
+      "Heartbreaker",
+      "Invincible",
+      "Break of Dawn",
+      "Heaven Can Wait",
+      "You Rock My World",
+      "Butterflies",
+      "Speechless",
+      "2000 Watts",
+      "You Are My Life",
+      "Privacy",
+      "Don't Walk Away",
+      "Cry",
+      "The Lost Children",
+      "Whatever Happens",
+      "Threatened"
+    ],
+    "bgColor": 215
+  },
+  {
+    "title": "Michael (2010)",
+    "cover": "michael",
+    "songs": [
+      "Hold My Hand",
+      "Hollywood Tonight",
+      "Keep Your Head Up",
+      "(I Like) The Way You Love Me",
+      "Monster",
+      "Best of Joy",
+      "Breaking News",
+      "(I Can't Make It) Another Day",
+      "Behind the Mask",
+      "Much Too Soon"
+    ],
+    "bgColor": 8
+  },
+  {
+    "title": "Xscape (2014)",
+    "cover": "xscape",
+    "songs": [
+      "Love Never Felt So Good",
+      "Chicago",
+      "Loving You",
+      "A Place with No Name",
+      "Slave to the Rhythm",
+      "Do You Know Where Your Children Are",
+      "Blue Gangsta",
+      "Xscape"
+    ],
+    "bgColor": 33
+  }
+];
+
+let doc = document.body;
+let navTab = document.querySelector(".navigation");
+let infoTab = document.querySelector(".info");
+let navTabButtonsArray;
+
+function fillNavTab() {
+  navTab.innerHTML = "";
+  let navTabButton;
```

Comment:
No need to reassign this variable. Just use `const` and use it inside `forEach`

Reply:
Don't understand about "reassign". 
navTabButton is a variable that I use to create DOM elements. Even if I use "forEach", not "for" - I need to put different content in it, so it can't be a const.

Reply:
It seems I didn't explain my thoughts exactly.
I meant no need to create this variable outside the `for` loop. 
```suggestion
for (let i = 0; i < albums.length; i++) {
    const navTabButton = document.createElement("button");
```

Reply:
Well, it's whole new treatment for me: instead of creating a variable one time and reuse it many times (let's call it "classic way") we re-create a constant for many times. Is it right? I've never seen such a treatment.

Reply:
I think it's no big deal. You can leave it as you wrote before. It will work fine. But!
As you use the `navTabButton` variable only inside the loop, it's better to use `const`. It helps you to prevent accidental reassignment of this variable. Also, it helps to keep your code clean and easy to read.

Reply:
Then I'll keep it in "classic way". 
May be, I'm too old :)
I promise not to reassign my variables, even accidentally.

Reply:
@ses3332002 I will add my opinion. It is not only about reassigning. In low level languages, as C/C++, this can be source of problems, if you will start counting assembler instructions, etc. But this is a JavaScript, which is quite high-level and compiler very smart - we have "warm up functions", we have JIT and all other magic necessary to load YouTube in 4k under 200 ms.

Also, you will not have a lot of data here. And in frontend in general. If you "really" will need to loop thru 10k elements, or 100 elements but 10-100 times, then "maybe" such approach will be reasonable. Especially for mobile phones. Or at backend.

But you should keep in mind, that you must avoid thing called "premature optimization". This approach in most cases usually creates bad code... and thats all it's result. In order to optimize something, you need to gather analytics in order to understand bottle necks. And then solve this bottle necks.

This is a just a variable and not a big problem. But try not to run before the train. :) Readability, clean code comes first.